### PR TITLE
Rename TypeProto nested messages

### PR DESCRIPTION
**Pull request by ezyang**
Fixes https://github.com/onnx/onnx/issues/78

### CODE REVIEW

File: onnx/onnx-ml.proto [hunk 1]
```diff
@@ -457,37 +441,31 @@ message ValueProto {
...
-  }
+// Define the types.
+message TypeProto {
 
-  message SparseTensorTypeProto {
+  message Tensor {
     // This field MUST NOT have the value of UNDEFINED
     // This field MUST be present for this version of the IR.
     optional TensorProto.DataType elem_type = 1;
     optional TensorShapeProto shape = 2;
```
Comment (NiklasGustafsson) [inline onnx/onnx-ml.proto]: Could we add a reserved properties block here, such as 100-200?

Comment (NiklasGustafsson) [reply onnx/onnx-ml.proto]: In fact, systematically reserving blocks everywhere should help future-proof, I believe.

**Comment by dzhulgakov** [comment]
Btw, does moving nested messages outside changes on-the-wire format?

**Comment by ezyang** [comment]
No; message names never occur on the wire.

**Comment by ezyang** [comment]
Rebased.

**Pull request merged**